src/components/PollViewer/PollViewer.vue

```diff
@@ -0,0 +1,425 @@
+		},
+
+		token() {
+			return this.activePoll?.token
+		},
+
+		container() {
+			return this.$store.getters.getMainContainerSelector()
+		},
+
+		poll() {
+			return this.$store.getters.getPoll(this.token, this.id)
+		},
+
+		selfHasVoted() {
+			return this.poll?.votedSelf?.length > 0
+		},
+
+		isPollPublic() {
+			return this.poll?.resultMode === POLL.MODE.PUBLIC
+		},
+
+		isPollOpen() {
+			return this.poll?.status === POLL.STATUS.OPEN
+		},
+
+		isPollClosed() {
+			return this.poll?.status === POLL.STATUS.CLOSED
+		},
+
+		isMultipleAnswers() {
+			return this.poll?.maxVotes === POLL.ANSWER_TYPE.MULTIPLE
+		},
+
+		checked: {
+			get() {
+				return this.voteToSubmit
+			},
+			set(value) {
+				this.voteToSubmit = Array.isArray(value) ? value : [value]
+			},
+		},
+
+		disabled() {
+			return this.loading || this.voteToSubmit.length === 0
+		},
+
+		selfIsOwnerOrModerator() {
+			return this.$store.getters.isModerator
+				|| (this.poll?.actorType === this.$store.getters.getActorType()
+					&& this.poll?.actorId === this.$store.getters.getActorId())
+		},
+
+		pollSummaryText() {
+			if (this.isPollClosed) {
+				return n('spreed', 'Poll results • %n vote', 'Poll results • %n votes', this.poll?.numVoters)
+			}
+
+			if (this.selfIsOwnerOrModerator || (this.isPollPublic && this.selfHasVoted)) {
+				return n('spreed', 'Open poll • %n vote', 'Open poll • %n votes', this.poll?.numVoters)
+			}
+
+			if (!this.isPollPublic && this.selfHasVoted) {
+				return t('spreed', 'Open poll • You voted already')
+			}
+
+			return t('spreed', 'Open poll')
+		},
+
+		canEndPoll() {
+			return this.isPollOpen && this.selfIsOwnerOrModerator
+		},
+	},
+
+	watch: {
+		modalPage(value) {
+			if (value === 'voting') {
+				this.setVoteData()
+			}
+		},
+
+		poll: {
+			immediate: true,
+			handler(value) {
+				if (!value) {
+					this.modalPage = ''
+				} else if (this.selfHasVoted || this.isPollClosed) {
+					this.modalPage = 'results'
+				} else {
+					this.modalPage = 'voting'
+				}
+			},
+		},
+	},
+
+	mounted() {
+		EventBus.$on('talk:poll-added', this.showModal)
+	},
+
+	beforeDestroy() {
+		EventBus.$off('talk:poll-added', this.showModal)
+	},
+
+	methods: {
+		getPollData() {
+			if (!this.poll) {
+				this.$store.dispatch('getPollData', {
+					token: this.token,
+					pollId: this.id,
+				})
+			}
+		},
+
+		setVoteData() {
+			this.voteToSubmit = this.selfHasVoted
+				? this.poll?.votedSelf.map(el => el.toString())
+				: []
+		},
+
+		showModal(payload) {
+			if (!this.activePoll && this.isInCall) {
+				this.$store.dispatch('setActivePoll', payload)
+			}
+		},
+
+		dismissModal() {
+			this.$store.dispatch('removeActivePoll')
+			this.voteToSubmit = []
+		},
+
+		async submitVote() {
+			this.loading = true
+			await this.$store.dispatch('submitVote', {
+				token: this.token,
+				pollId: this.id,
+				vote: this.voteToSubmit.map(element => +element),
+			})
+			this.modalPage = 'results'
+			this.loading = false
+		},
```

Comment:
Add error handling here. If there is an error, keep the voting page and showError

Reply:
You can add error handling in other requests where it is needed too.